jvb/src/main/kotlin/org/jitsi/videobridge/load_management/JvbLoadManager.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge.load_management
+
+import org.jitsi.nlj.util.NEVER
+import org.jitsi.utils.logging2.cdebug
+import org.jitsi.utils.logging2.createLogger
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
+import java.util.logging.Level
+
+class JvbLoadManager<T : JvbLoadMeasurement> @JvmOverloads constructor(
+    private val jvbLoadThreshold: T,
+    private val jvbRecoveryThreshold: T,
+    private val loadReducer: JvbLoadReducer,
+    private val clock: Clock = Clock.systemUTC()
+) {
+    private val logger = createLogger(minLogLevel = Level.ALL)
+
+    private var lastReducerTime: Instant = NEVER
+
+    fun loadUpdate(loadMeasurement: T) {
+        logger.cdebug { "Got a load measurement of $loadMeasurement" }
+        if (loadMeasurement.getLoad() >= jvbLoadThreshold.getLoad()) {
```

Comment:
A couple things I think are nice about having the separate `JvbLoadMeasurement` type:
1. It guarantees the LoadReducer, the measurement and the threshold are all in sync (since the LoadReducer is generic around the measurement type.  I also thought representing the threshold as a measurement made sense, as that's really all it is.
1. It allows for nicer logging of the measurement (since specific instances can override `toString`)
1. Arguably it _shouldn't_ always return a Double, and the type gives us more flexibility.  I could change the code to reflect this.

Open to discussion on it thought.

Reply:
It's probably fine for now.